src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java

```diff
@@ -0,0 +1,207 @@
+        Driver lewis = createDriverLewis();
+        Car monopost2021 = new Car();
+
+        monopost2021.setName("monopost 2021");
+        monopost2021.setDriver(lewis);
+        lewis.setCar(monopost2021);
+
+        driverDao.create(lewis);
+        carDao.create(monopost2021);
+
+        Car found = carDao.findById(monopost2021.getId());
+        Assert.assertEquals(found, monopost2021);
+    }
+
+    @Test
+    public void findByNameTest() {
+        Driver lewis = createDriverLewis();
+        Car monopost2021 = new Car();
+
+        monopost2021.setName("monopost 2021");
+        monopost2021.setDriver(lewis);
+        lewis.setCar(monopost2021);
+
+        driverDao.create(lewis);
+        carDao.create(monopost2021);
+
+        Car found = carDao.findByName(monopost2021.getName());
+        Assert.assertEquals(found, monopost2021);
+    }
+
+    @Test
+    public void findAllTest() {
+        Car car1 = new Car();
+        car1.setName("monopost1");
+        carDao.create(car1);
+        Assert.assertEquals(carDao.findAll().size(), 1);
+
+        Car car2 = new Car();
+        car2.setName("monopost2");
+        carDao.create(car2);
+        Assert.assertEquals(carDao.findAll().size(), 2);
+
+        Car car3 = new Car();
+        car3.setName("monopost3");
+        carDao.create(car3);
+        Assert.assertEquals(carDao.findAll().size(), 3);
+    }
+
+    @Test
+    public void updateCar() {
+        Car car = new Car();
+        car.setName("AMG Petronas 2020");
+
+        Driver lewis = createDriverLewis();
+        Driver fernando = createDriverFernando();
+
+        car.setDriver(lewis);
+        lewis.setCar(car);
+
+        driverDao.create(lewis);
+        driverDao.create(fernando);
+        carDao.create(car);
+
+        car.setName("AMG petronas ultra fast uber new car");
+        car.setDriver(fernando);
+        lewis.setCar(null);
+        fernando.setCar(car);
+
+        driverDao.update(lewis);
+        driverDao.update(fernando);
+        carDao.update(car);
+
+        Car found = carDao.findById(car.getId());
+        Assert.assertEquals(found.getName(), "AMG petronas ultra fast uber new car");
+        Assert.assertNull(lewis.getCar());
+        Assert.assertEquals(found.getDriver(), fernando);
+    }
+
+    @Test
+    public void removeCarTest() {
+        Car carFerrari = new Car();
+        carFerrari.setName("F1 Scuderia Ferrari 2021 - car1");
+        carDao.create(carFerrari);
+
+        Car carMercedes = new Car();
+        carMercedes.setName("Petronas Mercedes - car1");
+        carDao.create(carMercedes);
+
+        Assert.assertEquals(carDao.findAll().size(), 2);
+
+        carDao.remove(carFerrari);
+
+        Assert.assertEquals(carDao.findAll().size(), 1);
+    }
+
+    public Driver createDriverLewis() {
+        Driver driver = new Driver();
+        driver.setName("Lewis");
+        driver.setSurname("Hamilton");
+        driver.setNationality("GB");
+        driver.setAgresive(true);
```

Comment:
melo by byt setAggressive ... ve fci je typo 😄

Reply:
sorry, my bad. I have fixed it